### ISSUE TITLE
Object.dirname is deprecated.

### ISSUE THREAD

**Issue by Andretalik** [pchomik/linter-python#39]
Argument to `path.dirname` must be a string
```
Object.dirname (C:\Users\Gman\AppData\Local\atom\app-1.15.0\resources\app.asar\src\electron-shims.js:9:10)
PluginLinter.calculateProjectDir (C:\Users\Gman\.atom\packages\linter-python\lib\linter.js:73:28)
PluginLinter.lint (C:\Users\Gman\.atom\packages\linter-python\lib\linter.js:29:31)
promises.push.Promise.then._this2.emitter.emit.number (C:\Users\Gman\.atom\packages\linter\lib\linter-registry.js:141:28)
_loop (C:\Users\Gman\.atom\packages\linter\lib\linter-registry.js:139:25)
LinterRegistry.<anonymous> (C:\Users\Gman\.atom\packages\linter\lib\linter-registry.js:182:22)
```

**Comment by Arcanemagus** [comment]
@pchomik Looks like you need a check before [this line](https://github.com/pchomik/linter-python/blob/e7e56be9a73e6deb4242ab864b73046170914aa3/src/linter.ts#L43) to make sure that `filePath` isn't `undefined`.

(Also, there is no need to pass in a `TextEditor` to `calculateProjectDir` as you are just using it to get the same value being passed in as `filePath`.)

**Comment by pchomik** [comment]
Should be fixed by #40
